src/akamai/js/akamaiHD_flash.js

```diff
@@ -201,8 +211,29 @@ require("../../../html5-common/js/utils/constants.js");
      * @returns {boolean} True or false indicating success
      */
     this.setVideoUrl = function(url, encoding) {
-      /* Should be set to false if stream URL changes */
-      loaded = false;
+      newController=this.controller;
```

Comment:
alternatively why not use something like 
var self = this;
Then you would use self.controller in callbacks.  That'll give you access to all this.* functions